### DISCUSSION TITLE
Countdown Beeps

### DISCUSSION THREAD

**Discussion by ptojsl** [dvmarinoff/Auuki#125] [Ideas]
Hi,

I love the application! I used to use Trainerroad, and this app has helped me save so much money!

One thing I did love about trainerroad that this app doesn't have is a countdown of three beeps that lets you know when the next interval is coming up. 

Is there a possibility this could be added as a setting? I would imagine it would help other cyclists as well, since you don't have to stare at the Flux app all the time to see when the next interval comes up.

Thanks,
Peter

**Comment by dvmarinoff** [comment]
Yes, sure! Will put that on the queue and will notify you when the first version is out, if you wish to help with testing and refining.

One of the very first versions had:
Audio for desktop,
Audio or Vibrations for phones,

but it got lost somewhere during code refactoring.

**Comment by dvmarinoff** [comment]
@ptojsl just letting you know I added countdown beeps on the developer version of the app at: [flux-devel.vercel.app](https://flux-devel.vercel.app/). Volume controls are on the Settings page under the connection settings. 

I am using Web Audio API oscillator and a gain node to synth beep sounds. There are 4 beeps 150ms long, first 3 are C5 last one is A5.
The count with the timer changes at 00:03, 00:02, 00:01, 00:00. And will execute on each interval that is at least 4 seconds long.

This is first iteration and there is high possibility for bugs and weird behaviour.

**Comment by dvmarinoff** [comment]
Just merged the countdown beeps to master. The feature can be now used on the main version as well.

**Comment by ptojsl** [comment]
Hi Dimitar,

Thank you so much. Your work is unbelievable.

Sincerely,
Peter
